lib/modules/manager/azure-pipelines/extract.ts

```diff
@@ -83,18 +93,67 @@ export function extractPackageFile(
   }
 
   // grab the repositories tags
-  for (const repository of pkg.resources.repositories) {
-    const dep = extractRepository(repository);
-    if (dep) {
-      deps.push(dep);
+  if (
+    !is.nullOrUndefined(pkg.resources) &&
+    is.nonEmptyArray(pkg.resources.repositories)
+  ) {
+    for (const repository of pkg.resources.repositories) {
+      const dep = extractRepository(repository);
+      if (dep) {
+        deps.push(dep);
+      }
     }
   }
 
   // grab the containers tags
-  for (const container of pkg.resources.containers) {
-    const dep = extractContainer(container);
-    if (dep) {
-      deps.push(dep);
+  if (
+    !is.nullOrUndefined(pkg.resources) &&
+    is.nonEmptyArray(pkg.resources.containers)
+  ) {
+    for (const container of pkg.resources.containers) {
+      const dep = extractContainer(container);
+      if (dep) {
+        deps.push(dep);
+      }
+    }
+  }
```

Comment:
```suggestion 
  for (const container of pkg.resources?.containers ?? []) {
    const dep = extractContainer(container);
    if (dep) {
      deps.push(dep);
    }
  }
```

reduce nesting and simplify code?